cpp/src/arrow/tensor.cc

```diff
@@ -269,13 +270,15 @@ struct ConvertColumnsToTensorRowMajorVisitor {
       auto in_values = ArraySpan(in_data).GetSpan<In>(1, in_data.length);
 
       if (in_data.null_count == 0) {
-        for (int64_t i = 0; i < in_data.length; ++i) {
-          out_values[i * num_cols + col_idx] = static_cast<Out>(in_values[i]);
+        for (int64_t data_idx = 0; data_idx < in_data.length; ++data_idx) {
```

Comment:
It might be fine to keep `i` here as the loop variable, since it is a short loop and clear where the value is coming from (i.e. iterating over length of `in_data`), that keeps the line a bit shorter